cgen error: incompatible types when assigning to type ‘main__Server’ from type ‘main__Server *’

### Describe the bug

happens when call `spawn server.client_count_monitor()`

### Reproduction Steps

```sh
 clear && v wipe-cache && v -prod crun ./c/10_circular_queue.v 
```

```v
/*
wrk -t16 -c512 -d60s http://127.0.0.1:8080

Running 1m test @ http://127.0.0.1:8080
  16 threads and 512 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     3.52ms    1.78ms  27.50ms   78.20%
    Req/Sec     6.90k     1.55k    8.83k    85.70%
  6597405 requests in 1.00m, 755.01MB read
Requests/sec: 109780.45
Transfer/sec:     12.56MB
*/
import sync
import time

#include <arpa/inet.h>

pub enum SocketType {
	udp       = C.SOCK_DGRAM
	tcp       = C.SOCK_STREAM
	seqpacket = C.SOCK_SEQPACKET
}

// AddrFamily are the available address families
pub enum AddrFamily {
	unix   = C.AF_UNIX
	ip     = C.AF_INET
	ip6    = C.AF_INET6
	unspec = C.AF_UNSPEC
}

fn C.socket(domain AddrFamily, typ SocketType, protocol int) int

fn C.htons(host u16) u16

fn C.bind(sockfd int, address &C.sockaddr_in, addrlen u32) int

fn C.listen(sockfd int, backlog int) int

fn C.accept(sockfd int, address &C.sockaddr_in, addrlen &u32) int

fn C.setsockopt(sockfd int, level int, optname int, optval &int, optlen u32) int

fn C.recv(sockfd int, buffer &u8, len u32, flags int) int

fn C.send(sockfd int, buffer &u8, len u32, flags int) int

pub struct C.sockaddr_in {
mut:
	sin_family AddrFamily
	sin_port   u16
	sin_addr   int
}

const port = 8080
const max_waiting_queue_size = 512
const backlog = 512
const buffer_size = 140
const response_body = '{ "message": "Hello, world!" }'
const response = 'HTTP/1.1 200 OK\r\nContent-Type: application/json\r\nContent-Length: ${response_body.len}\r\nConnection: close\r\n\r\n${response_body}'

struct Server {
mut:
	server_socket    int
	client_queue     [max_waiting_queue_size]int
	queue_head       int
	queue_tail       int
	queue_lock       sync.Mutex
	client_count     u32
	has_clients      bool
	thread_pool_size u32 = u32(8)
	shutdown_flag    bool
}

// Enqueues a client connection
fn (mut s Server) enqueue_client(client int) {
	s.queue_lock.lock()
	if s.client_count < max_waiting_queue_size {
		s.client_queue[s.queue_tail] = client
		s.queue_tail = (s.queue_tail + 1) % max_waiting_queue_size
		s.client_count++
		s.has_clients = true
	} else {
		C.close(client)
	}
	s.queue_lock.unlock()
}

// Dequeues a client connection
fn (mut s Server) dequeue_client() int {
	mut client := -1
	for {
		s.queue_lock.lock()
		if s.client_count > 0 {
			client = s.client_queue[s.queue_head]
			s.queue_head = (s.queue_head + 1) % max_waiting_queue_size
			s.client_count--
			if s.client_count == 0 {
				s.has_clients = false
			}
			s.queue_lock.unlock()
			break
		}
		s.queue_lock.unlock()
		time.sleep(100 * time.microsecond)
	}
	return client
}

// Worker thread function
fn (mut s Server) worker_thread() {
	for {
		if s.shutdown_flag {
			break
		}
		client := s.dequeue_client()
		if client < 0 {
			continue
		}
		buffer := [buffer_size]u8{}
		if C.recv(client, &buffer[0], buffer_size, 0) <= 0 {
			C.close(client)
			continue
		}
		C.send(client, response.str, response.len, 0)
		C.close(client)
	}
}

fn (server Server) client_count_monitor() {
	for {
		if server.shutdown_flag {
			break
		}
		println('Client count: ${server.client_count}')
		time.sleep(1 * time.second)
	}
}

fn main() {
	mut server := &Server{
		server_socket: C.socket(AddrFamily.ip, SocketType.tcp, 0)
	}

	if server.server_socket < 0 {
		eprintln('socket failed')
		return
	}

	opt := 1
	C.setsockopt(server.server_socket, C.SOL_SOCKET, C.SO_REUSEADDR | C.SO_REUSEPORT,
		&opt, sizeof(opt))

	address := C.sockaddr_in{
		sin_family: AddrFamily.ip
		sin_port:   C.htons(port)
		sin_addr:   C.INADDR_ANY
	}
	if C.bind(server.server_socket, &address, sizeof(address)) < 0
		|| C.listen(server.server_socket, backlog) < 0 {
		eprintln('bind or listen failed')
		return
	}


	println('Listening on http://localhost:${port}')

	for i := 0; i < server.thread_pool_size; i++ {
		spawn server.worker_thread()
	}
	spawn server.client_count_monitor()

	mut addrlen := sizeof(address)
	for {
		client := C.accept(server.server_socket, &address, &addrlen)
		if client < 0 {
			eprintln('accept failed')
			continue
		}
		server.enqueue_client(client)
	}
}
```

### Expected Behavior

Run as without `spawn server.client_count_monitor()` but printing each second

### Current Behavior

```sh
V cache folder /home/hitalo/.vmodules/.cache was wiped.
V tmp.c and tests folder folder /tmp/v_1000 was wiped.
================== C compilation error (from cc): ==============
cc:       |                                                 ^
cc: /tmp/v_1000/10_circular_queue.01JBJ3K79873WAVG7DZPB2BMHN.tmp.c:19812:25: error: incompatible types when assigning to type ‘main__Server’ from type ‘main__Server *’
cc: 19812 |         arg__t1->arg0 = server;
cc:       |                         ^~~~~~
cc: /tmp/v_1000/10_circular_queue.01JBJ3K79873WAVG7DZPB2BMHN.tmp.c:19841:60: warning: passing argument 2 of ‘accept’ from incompatible pointer type [-Wincompatible-pointer-types]
cc: 19841 |                 int client = accept(server->server_socket, &address, &addrlen);
cc:       |                                                            ^~~~~~~~
cc:       |                                                            |
cc:       |                                                            struct sockaddr_in *
cc: /usr/include/x86_64-linux-gnu/sys/socket.h:306:45: note: expected ‘struct sockaddr * restrict’ but argument is of type ‘struct sockaddr_in *’
cc:   306 | extern int accept (int __fd, __SOCKADDR_ARG __addr,
cc:       |                                             ^
... (the original output was 23 lines long, and was truncated to 12 lines)
================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.8 ee4f29f

### Environment details (OS name and version, etc.)

Linux

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzIzZTdhMzI3Y2M3NjgxYzYyNTdjN2UiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.v-8FtyFcr-uQAQ2ViJ-YWKUvm73h41ZA2BwO95o8GXc">Huly&reg;: <b>V_0.6-21165</b></a></sub>

--- a/vlib/v/gen/c/spawn_and_go.v
+++ b/vlib/v/gen/c/spawn_and_go.v
@@ -101,6 +101,10 @@ fn (mut g Gen) spawn_and_go_expr(node ast.SpawnExpr, mode SpawnGoMode) {
 	}
 	if expr.is_method {
 		g.write('${arg_tmp_var}${dot}arg0 = ')
+		if expr.left is ast.Ident && expr.left.obj.typ.is_ptr()
+			&& !node.call_expr.receiver_type.is_ptr() {
+			g.write('*')
+		}
 		g.expr(expr.left)
 		g.writeln(';')
 	}

--- a/vlib/v/tests/concurrency/spawn_with_different_method_receivers_test.v
+++ b/vlib/v/tests/concurrency/spawn_with_different_method_receivers_test.v
@@ -1,0 +1,23 @@
+struct Server {}
+
+fn (server Server) non_mutable_receiver() {
+	println(@LOCATION)
+}
+
+fn (mut s Server) mutable_receiver() {
+	println(@LOCATION)
+}
+
+fn (s &Server) reference_receiver() {
+	println(@LOCATION)
+}
+
+fn test_spawning_threads_with_methods_that_have_mutable_and_non_mutable_receivers() {
+	mut server := &Server{}
+	t1 := spawn server.non_mutable_receiver()
+	t2 := spawn server.mutable_receiver()
+	t3 := spawn server.reference_receiver()
+	t1.wait()
+	t2.wait()
+	t3.wait()
+}

